src/sqlfluff/core/parser/segments/base.py

```diff
@@ -343,7 +343,7 @@ def raw_segments(self) -> List["BaseSegment"]:
         return self.get_raw_segments()
 
     @cached_property
-    def raw_segments_upper(self) -> Optional[str]:
+    def first_non_whitespace_segment_raw_upper(self) -> Optional[str]:
```

Comment:
This property was poorly named. Updated the name to better reflect what it does.

Reply:
Does this not also need changed on line 666 (the sign of the devil!) to ensure the cached version is recalculated?

Reply:
Great catch! I used an automated tool to rename this, and it only updated the actual code.

If we added or changed cached fields very often, it'd probably be worthwhile to compute this list automatically. It's pretty rare, though.

Reply:
Fixed.